cmd/nvidia-ctk-installer/container/runtime/containerd/config_test.go

```diff
@@ -0,0 +1,978 @@
+            BinaryName = "/usr/bin/nvidia-container-runtime.cdi"
+            SystemdCgroup = true
+
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia-legacy]
+          container_annotations = ["cdi.k8s.io*"]
+          runtime_type = "io.containerd.runc.v2"
+
+          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia-legacy.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime.legacy"
+            SystemdCgroup = true
+
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          runtime_type = "io.containerd.runc.v2"
+
+          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+            BinaryName = "/usr/bin/runc"
+            SystemdCgroup = true
+
+    [plugins."io.containerd.grpc.v1.cri".registry]
+
+      [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+
+        [plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
+          endpoint = ["https://registry-1.docker.io"]
+
+  [plugins."io.containerd.internal.v1.opt"]
+    path = "/opt/containerd"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+			assertCleanupPostConditions: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+
+				if _, err := os.Stat(configPath); os.IsNotExist(err) {
+					return nil
+				}
+
+				// If file exists, verify nvidia runtimes were removed
+				actual, err := os.ReadFile(configPath)
+				require.NoError(t, err)
+				contentStr := string(actual)
+
+				require.NotContains(t, contentStr, "nvidia")
+				require.NotContains(t, contentStr, "nvidia-cdi")
+				require.NotContains(t, contentStr, "nvidia-legacy")
+				return nil
+			},
+		},
+		{
+			description:   "v2: existing config without default runtime (SetAsDefault=false)",
+			configVersion: 2,
+			containerOptions: container.Options{
+				Config:         "{{ .testRoot }}/etc/containerd/config.toml",
+				RuntimeName:    "nvidia",
+				RuntimeDir:     "/usr/bin",
+				SetAsDefault:   false,
+				RestartMode:    "none",
+				ExecutablePath: "not-containerd",
+			},
+			options: Options{},
+			prepareEnvironment: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+				require.NoError(t, os.MkdirAll(filepath.Dir(configPath), 0755))
+
+				// Write initial config without any default runtime
+				initialConfig := `version = 2
+
+[plugins]
+  [plugins."io.containerd.grpc.v1.cri"]
+    [plugins."io.containerd.grpc.v1.cri".containerd]
+      # Note: No default_runtime_name specified
+
+      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          runtime_type = "io.containerd.runc.v2"
+
+          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+            BinaryName = "/usr/bin/runc"
```

Comment:
Once again, we're actually interested in the runtime options from the `default_runtime_name`, `runc`, and `crun`.